soh/soh/Enhancements/randomizer/ShuffleScrubs.cpp

```diff
@@ -0,0 +1,32 @@
+#include "soh/Enhancements/game-interactor/GameInteractor.h"
+#include "soh/Enhancements/randomizer/context.h"
+#include "soh/Enhancements/randomizer/randomizerTypes.h"
+#include "soh/OTRGlobals.h"
+#include "soh/ShipInit.hpp"
+
+// RANDOTODO: Convert the reset of Scrub Shuffle to be in this file and
+// use hooks appropriately
+
+void RegisterShuffleScrubs() {
+    SHOULD_SHUFFLE_LOCATION({
+        if (location->GetRCType() == RCTYPE_SCRUB) {
+            switch (RAND_GET_OPTION(RSK_SHUFFLE_SCRUBS)) {
+                case RO_SCRUBS_OFF:
+                    *should = false;
+                    break;
+                case RO_SCRUBS_ONE_TIME_ONLY:
+                    switch (location->GetRandomizerCheck()) {
+                        case RC_LW_DEKU_SCRUB_GROTTO_FRONT:
+                        case RC_LW_DEKU_SCRUB_NEAR_BRIDGE:
+                        case RC_HF_DEKU_SCRUB_GROTTO:
```

Comment:
```suggestion
                        case RC_HF_DEKU_SCRUB_GROTTO:
                            *should = true;
```
need this no? I'm confused what not setting `*should` does since everything is wrapped in checking RCTYPE first, where if RCTYPE mismatch happens we don't touch `*should` so I'd expect default is false

Reply:
*should defaults to true, you only need to change it if it should be false. I did it that way on purpose so if you try to do a new shuffle, it's true and all the basic checks that we always shuffle don't need any hooks. Some of the ones I converted first set *should to a boolean expression, it's just that setting it to true is effectively "doing nothing" in those cases. These are what I meant when I said I might go back and clean some of those up a bit more.